src/Compilers/Core/Portable/Operations/IOperation.ChildCollection.Reversed.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public partial interface IOperation
+    {
+        /// <summary>
+        /// Implements a struct-based collection of <see cref="Operation"/> nodes. This collection is ordered, but
+        /// random access into the collection is not provided. This type is not hardened to 
+        /// <code>default(Enumerable)</code>, and will null reference in these cases.
+        /// </summary>
+        public readonly partial struct ChildCollection
+        {
+            [NonDefaultable]
+            public readonly struct Reversed : IReadOnlyCollection<IOperation>
```

Comment:
Note: `NonDefaultableAttribute` is used by an internal analyzer. This attribute has no effect for consumers outside Roslyn. Consider doc'ing that consumers shouldn't use `default(IOperation.ChildCollection.Reversed)` or something?